src/datareader.cpp

```diff
@@ -202,6 +208,25 @@ void DataReader::readFile(QString file)
             m_tracknum = QString::number(tagFile->tag()->track());
 
             if (m_title=="") m_title = QFileInfo(file).baseName();
+
+            // if we have artist and album, we check for a cover image.
+            if (m_artist != "" && m_album != "") {
+                QFileInfo info(file);
+                QDirIterator iterator(info.dir().path(), QDirIterator::Subdirectories);
+                while (iterator.hasNext()) {
+                    iterator.next();
+                    if (!iterator.fileInfo().isDir()) {
```

Comment:
Maybe `isFile()` instead of not `isDir()` ?

Reply:
You mean, swap from negating on dir to affirming on isFile?